sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_custom_form_client.py

```diff
@@ -0,0 +1,362 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+# pylint: disable=protected-access
+
+from typing import (  # pylint: disable=unused-import
+    Optional,
+    Any,
+    IO,
+    Iterable,
+    TYPE_CHECKING,
+)
+import six
+from azure.core.tracing.decorator import distributed_trace
+from azure.core.polling import LROPoller
+from azure.core.polling.base_polling import LROBasePolling  # pylint: disable=no-name-in-module,import-error
+from azure.core.exceptions import HttpResponseError
+from ._generated.models import AnalyzeOperationResult, Model
+from ._generated._form_recognizer_client import FormRecognizerClient as FormRecognizer
+from ._generated.models import TrainRequest, TrainSourceFilter
+from ._policies import CognitiveServicesCredentialPolicy
+from ._response_handlers import (
+    prepare_unlabeled_result,
+    prepare_labeled_result,
+)
+from ._helpers import get_content_type, POLLING_INTERVAL
+from ._models import (
+    ModelInfo,
+    ModelsSummary,
+    CustomModel,
+    CustomLabeledModel,
+)
+if TYPE_CHECKING:
+    from ._credential import FormRecognizerApiKeyCredential
+
+
+class CustomFormClient(object):
+    """CustomFormClient.
+
+    :param str endpoint: Supported Cognitive Services endpoints (protocol and hostname,
+        for example: https://westus2.api.cognitive.microsoft.com).
+    :param credential: Credentials needed for the client to connect to Azure.
+        This is an instance of FormRecognizerApiKeyCredential if using an API key.
+    :type credential: ~azure.ai.formrecognizer.FormRecognizerApiKeyCredential
+    """
+
+    def __init__(self, endpoint, credential, **kwargs):
+        # type: (str, FormRecognizerApiKeyCredential, Any) -> None
+        self._client = FormRecognizer(
+            endpoint=endpoint,
+            credential=credential,
+            authentication_policy=CognitiveServicesCredentialPolicy(credential),  # TODO: replace with core policy
+            **kwargs
+        )
+
+    @distributed_trace
+    def begin_training(self, source, source_prefix_filter="", include_sub_folders=False, **kwargs):
+        # type: (str, Optional[str], Optional[bool], Any) -> LROPoller
+        """Create and train a custom model. The request must include a source parameter that is an
+        externally accessible Azure storage blob container Uri (preferably a Shared Access Signature Uri).
+        Models are trained using documents that are of the following content type - 'application/pdf',
+        'image/jpeg', 'image/png', 'image/tiff'. Other type of content is ignored.
```

Comment:
What does "ignored" mean in this case? If there are only documents with content type image/bmp in the container, what failure will I see (if any)?

Reply:
I copied these docstrings from the swagger, still need to update them, but you do bring up an interesting scenario that needs clarity. Let me test this and see if we get a failure.

Reply:
So if there are only documents with an unsupported content-type in the container, I get back a model with status "invalid" and an error message saying there weren't any valid file types. If I mix valid content-types with invalid content-types in the container, I get back a valid model and the unsupported content is ignored.